launch/launch/actions/opaque_coroutine.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the OpaqueCoroutine action."""
+
+import asyncio
+import collections.abc
+from typing import Any
+from typing import Coroutine
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+
+from ..action import Action
+from ..event import Event
+from ..event_handlers import OnShutdown
+from ..launch_context import LaunchContext
+from ..some_actions_type import SomeActionsType
+from ..utilities import ensure_argument_type
+
+
+class OpaqueCoroutine(Action):
+    """
+    Action that adds a Python coroutine to the launch run loop.
+
+    The signature of a coroutine should be:
+
+    .. code-block:: python
+
+        async def coroutine(
+            context: LaunchContext,  # iff ignore_context is False
```

Comment:
I understand what `iff` is in this context, but it may be helpful to be a little bit more explicit.  It may also be worth updating the docs for `OpaqueFunction` so that they are consistent in this regard.

Reply:
That makes sense. I've updated the documentation, but I'm not sure what you mean about changing `OpaqueFunction` docs.